### ISSUE TITLE
s3: get_bucket_location doesn't return expected results

### ISSUE THREAD

**Issue by phyber** [awslabs/aws-sdk-rust#116]
### Version

```
├── aws-sdk-s3 v0.0.8-alpha (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   ├── aws-auth v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   ├── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   │   ├── smithy-types v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   ├── aws-endpoint v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   ├── aws-types v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   └── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   ├── aws-http v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   ├── aws-types v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── smithy-types v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   ├── aws-hyper v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   ├── aws-auth v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── aws-endpoint v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── aws-http v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── aws-sig-auth v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   │   ├── aws-auth v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   │   ├── aws-types v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   │   ├── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── protocol-test-helpers v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   ├── smithy-client v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   │   ├── protocol-test-helpers v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   │   ├── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   │   ├── smithy-http-tower v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)
│   │   │   │   ├── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   │   ├── smithy-types v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── smithy-http-tower v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   │   ├── smithy-types v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   ├── aws-sig-auth v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   ├── aws-types v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   ├── smithy-http v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   ├── smithy-types v0.0.1 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939) (*)
│   └── smithy-xml v0.1.0 (https://github.com/awslabs/aws-sdk-rust?tag=v0.0.8-alpha#2d876939)

```
### Platform

FreeBSD and Linux

### AWS Services

S3

### Description

When using `get_bucket_location` the returned location constrained always seems to be `None`, even for buckets outside of `us-east-1`.

The following code should demonstrate it. I was running this in `eu-west-1` with buckets also residing in `eu-west-1`

```rust
// Excuse my unwraps, it's just demo code.
use aws_sdk_s3::Client;

#[tokio::main]
async fn main() {
    let client = Client::from_env();
    let resp = client.list_buckets()
        .send()
        .await
        .unwrap();

    // Get a list of buckets
    let bucket_names = if let Some(buckets) = resp.buckets {
        buckets
            .iter()
            .filter_map(|b| b.name.to_owned())
            .collect()
    }
    else {
        Vec::new()
    };

    // Try to get their locations
    for bucket in bucket_names {
        let resp = client.get_bucket_location()
            .bucket(&bucket)
            .send()
            .await
            .unwrap();

        // All of the location constraints are None
        println!(
            "Bucket {} has location_constraint: {:?}",
            &bucket,
            resp.location_constraint,
        );
    }
}
```

**Comment by jdisanti** [comment]
Thanks for submitting this and providing the example code!

I have reproduced the issue locally and will work on a fix.

**Comment by phyber** [comment]
Excellent, thanks. I was hoping I wasn't just holding it incorrectly 😅

**Comment by rcoh** [comment]
For anyone else who might be reading this, I just want to say that even if you _were_ holding it incorrectly, in our eyes, that is still our responsibility to fix. We are trying to build an SDK that isn't just correct, but obvious / hard to misuse. So please file issues for problems that occurred due to misuse or confusion, we want to hear them!

**Comment by jdisanti** [comment]
The [fix for this](https://github.com/awslabs/smithy-rs/pull/516) has landed in smithy-rs and will go out with the next release.

**Comment by phyber** [comment]
Excellent, thank you very much.

**Comment by jdisanti** [comment]
This is fixed in `v0.0.9-alpha`.
